### PR TITLE
Copybook tracker

### PR DESCRIPTION
**Pull request by zacanbrcom**
Address the issue #180 and extends the #141.
This POC consider a very basic scenario and didn't include a proper definition of ```setting.json``` that may be introduced later on.

**End user scenario**
1. Cobol file opened with a missing copybook
2. the related dependency file is generated
3. The copybook tracker extension ask user credential to connect to mf and download the copybook into the .copybooks folder
5. The LSP trigger a new analysis for the cobol file and the syntax error of copybook not found disappear.

### CODE REVIEW

**Comment by ishche** [comment]
Pls notice that this version has done without zowe imperatives dependency

File: clients/cobol-lsp-vscode-extension/src/CopybooksDownloader.ts [hunk 1]
```diff
@@ -0,0 +1,109 @@
...
+        const items: vscode.QuickPickItem[] = profiles.map(e => {
+            return {
+                description: e.username + "@" + e.host + ":" + e.port,
+                label: e.name,
+                picked: e.default,
+                value: e,
+            };
+        });
+
+        // FIXME rewrite without value 'hack'
```
Comment (zacanbrcom) [inline clients/cobol-lsp-vscode-extension/src/CopybooksDownloader.ts]: Could we get rid of this?

File: clients/cobol-lsp-vscode-extension/src/extension.ts [hunk 2]
```diff
@@ -56,18 +60,31 @@ export async function activate(context: ExtensionContext) {
...
-    item.text = "Language Server is active";
+function depChange(uri: Uri, downloader: CopybooksDownloader) {
+    const copybooks: string[] = fs.readFileSync(uri.fsPath).toString().split("\n")
+        .filter(e => e.trim().length > 0)
+        .map(e => e.trim());
+    downloader.downloadCopyBooks(copybooks);
+}
+
+function initWorkspaceTracker(downloader: CopybooksDownloader): Disposable {
+    const watcher = workspace.createFileSystemWatcher("**/" + DEPENDENCIES_FOLDER + "/**.deps", false, false, true);
```
Comment (zacanbrcom) [inline clients/cobol-lsp-vscode-extension/src/extension.ts]: the file extension generate from LSP server for dependency file is **.dep**

**Comment by ishche** [comment]
This is a wrong branch

**Pull request closed without merging**